rust/lance-namespace/src/dir.rs

```diff
@@ -0,0 +1,680 @@
+                    
+                    // Add storage options from config
+                    for (key, value) in config.storage_options() {
+                        operator_config.insert(key.clone(), value.clone());
+                    }
+                    
+                    let op = Operator::via_iter(opendal::Scheme::S3, operator_config)?;
+                    Ok(op)
+                }
+                "gcs" => {
+                    let bucket = url.host_str().ok_or_else(|| {
+                        LanceNamespaceError::InvalidConfiguration("GCS URL must have a bucket".to_string())
+                    })?;
+                    
+                    operator_config.insert("bucket".to_string(), bucket.to_string());
+                    operator_config.insert("root".to_string(), url.path().trim_start_matches('/').to_string());
+                    
+                    // Add storage options from config
+                    for (key, value) in config.storage_options() {
+                        operator_config.insert(key.clone(), value.clone());
+                    }
+                    
+                    let op = Operator::via_iter(opendal::Scheme::Gcs, operator_config)?;
+                    Ok(op)
+                }
+                "azblob" => {
+                    // For Azure Blob Storage, we might need to use "abs" or similar
+                    // Let's try to create it and catch the error if the scheme is not supported
+                    let container = url.host_str().ok_or_else(|| {
+                        LanceNamespaceError::InvalidConfiguration("Azure Blob URL must have a container".to_string())
+                    })?;
+                    
+                    operator_config.insert("container".to_string(), container.to_string());
+                    operator_config.insert("root".to_string(), url.path().trim_start_matches('/').to_string());
+                    
+                    // Add storage options from config  
+                    for (key, value) in config.storage_options() {
+                        operator_config.insert(key.clone(), value.clone());
+                    }
+
+                    let op = Operator::via_iter(opendal::Scheme::Azblob, operator_config)?;
+                    Ok(op)
+                }
+                "fs" => {
+                    operator_config.insert("root".to_string(), url.path().to_string());
+                    
+                    // Add storage options from config
+                    for (key, value) in config.storage_options() {
+                        operator_config.insert(key.clone(), value.clone());
+                    }
+                    
+                    let op = Operator::via_iter(opendal::Scheme::Fs, operator_config)?;
+                    Ok(op)
+                }
+                _ => {
+                    // Default fallback - treat as root with storage options
```

Comment:
default fallback should just use whatever scheme that is provided to initialize the operator with storage options.